src/server.rs

```diff
@@ -1,110 +1,100 @@
 use crate::client::ExecutionClient;
-use crate::debug_api;
+use crate::debug_api::DebugServer;
 use crate::metrics::ServerMetrics;
 use alloy_primitives::B256;
-use debug_api::DebugServer;
-use std::num::NonZero;
+use moka::sync::Cache;
+use parking_lot::Mutex;
 use std::sync::Arc;
 use std::time::Instant;
 
 use alloy_rpc_types_engine::{
     ExecutionPayload, ExecutionPayloadV3, ForkchoiceState, ForkchoiceUpdated, PayloadId,
     PayloadStatus,
 };
-use jsonrpsee::core::{async_trait, ClientError, RegisterMethodError, RpcResult};
+use jsonrpsee::RpcModule;
+use jsonrpsee::core::{ClientError, RegisterMethodError, RpcResult, async_trait};
 use jsonrpsee::types::error::INVALID_REQUEST_CODE;
 use jsonrpsee::types::{ErrorCode, ErrorObject};
-use jsonrpsee::RpcModule;
-use lru::LruCache;
 use op_alloy_rpc_types_engine::{OpExecutionPayloadEnvelopeV3, OpPayloadAttributes};
 use opentelemetry::global::{self, BoxedSpan, BoxedTracer};
 use opentelemetry::trace::{Span, TraceContextExt, Tracer};
 use opentelemetry::{Context, KeyValue};
 use serde::{Deserialize, Serialize};
 
-use tokio::sync::Mutex;
 use tracing::{debug, error, info};
 
 use jsonrpsee::proc_macros::rpc;
 
-const CACHE_SIZE: usize = 100;
+const CACHE_SIZE: u64 = 100;
 
 pub struct PayloadTraceContext {
-    tracer: Arc<BoxedTracer>,
-    block_hash_to_payload_ids: Arc<Mutex<LruCache<B256, Vec<PayloadId>>>>,
-    payload_id_to_span: Arc<Mutex<LruCache<PayloadId, Arc<BoxedSpan>>>>,
-    local_to_external_payload_ids: Arc<Mutex<LruCache<PayloadId, PayloadId>>>,
+    tracer: BoxedTracer,
+    block_hash_to_payload_ids: Cache<B256, Vec<PayloadId>>,
```

Comment:
curious how this Cache differs from LruCache?

Reply:
It just lock free, which means we don't need to worry about locking ourselves. Helps avoid errors! Should be significantly faster as well, not that it really matters in this case haha.